lima, limactl: add page (#10648)
tldr-pages/tldr: pages/linux/limactl.md

> # lima

Reviewer: The command is `limactl` itself, not `lima`.
```suggestion
# limactl
```

> > Virtual machine manager for linux guests, with multiple vm templates available.

Reviewer 2: ```suggestion
> Virtual machine manager for Linux guests, with multiple VM templates available.
```

> > Can be used to run containers on macos, but also for generic virtual machine use-cases on macos and linux hosts.

Reviewer 2: ```suggestion
> Can be used to run containers on macos, but also for generic virtual machine use-cases on macOS and Linux hosts.
```

> > Can be used to run containers on macos, but also for generic virtual machine use-cases on macOS and Linux hosts.

Reviewer 3: ```suggestion
> Can be used to run containers on macOS, but also for generic virtual machine use cases on macOS and Linux hosts.
```

> More information: <https://github.com/lima-vm/lima>.

- Pick any of the available distributions, see templates in `/usr/local/share/lima/templates/`:

> `LIMA_DISTRO=debian|fedora|ubuntu|…`

Reviewer 2: ```suggestion
`LIMA_DISTRO={{debian|fedora|ubuntu|…}}`
```
We use `{{}}` in our pages so that our clients can highlight them as placeholders.

> - Pick any of the available distributions, see templates in `/usr/local/share/lima/templates/`:
> 
> `LIMA_DISTRO={{debian|fedora|ubuntu|…}}`
> 

Reviewer 3: ```suggestion
```

> - Create a virtual machine using the default settings (this will download the vm image on the first run):

Reviewer 2: ```suggestion
- Create a virtual machine using the default settings (this will download the VM image on the first run):
```

> - Create a virtual machine using the default settings (this will download the VM image on the first run):
> 
> `cat /usr/local/share/lima/templates/$LIMA_DISTRO.yaml | limactl create --name=$LIMA_DISTRO-vm -`

Reviewer 3: ```suggestion
- Create a virtual machine using the default settings (this will open a TUI to change the template and other settings):

`limactl create --name {{vm_name}}`
```

> - Start the vm instance (this might install some dependencies in the vm and take a few minutes):

Reviewer 2: ```suggestion
- Start the VM instance (this might install some dependencies in the VM and take a few minutes):
```

`limactl start $LIMA_DISTRO-vm`

- Open a remote shell inside the VM:

`limactl shell $LIMA_DISTRO-vm`

> - Stop/shutdown the vm instance:

Reviewer 2: ```suggestion
- Stop/shutdown the VM instance:
```

`limactl stop $LIMA_DISTRO-vm`

> - Delete the vm instance:

Reviewer 2: ```suggestion
- Delete the VM instance:
```

> - Start the VM instance (this might install some dependencies in the VM and take a few minutes):
> 
> `limactl start $LIMA_DISTRO-vm`
> 
> - Open a remote shell inside the VM:
> 
> `limactl shell $LIMA_DISTRO-vm`
> 
> - Stop/shutdown the VM instance:
> 
> `limactl stop $LIMA_DISTRO-vm`
> 
> - Delete the VM instance:
> 
> `limactl remove $LIMA_DISTRO-vm`

Reviewer 3: ```suggestion
- List VM instances:

`limactl list`

- Start the VM instance (this might install some dependencies in the VM and take a few minutes):

`limactl start {{vm_name}}`

- Open a remote shell inside the VM:

`limactl shell {{vm_name}}`

- Run a command inside the VM:

`limactl shell {{vm_name}} {{command}`

- Stop/shutdown the VM instance:

`limactl stop {{vm_name}}`

- Delete the VM instance:

`limactl remove {{vm_name}}`
```

Author: Hm.. I see this is opinionated, but I think that using the environment variable for selecting the distro and using that in the vm name is a neat trick that I find useful and wanted to share here. In my opinion, this adds additional value to the cheatsheet compared to the output of `limactl --help` (which I found a bit suboptimal which was my motivation to open this pr).

Author: In my opinion, the great thing about tldr pages is that it gives me a quick start with clis often when I am fine with opinionated examples.. for cases where this is not suitable for my needs I need to do further research.
